### PR TITLE
[favourites][listproviders] Add support for Video Select Actions for Favourites

### PR DESCRIPTION
**Pull request by ksooo**
Follow-up to #23848, bringing the missing bits 'n pieces for Favourites, namely full support for video select actions.

Runtime-tested on macOS and Android, latest Kodi master.

@enen92 should be straight-forward to review.

### CODE REVIEW

File: xbmc/FileItem.cpp [hunk 1]
```diff
@@ -3766,6 +3766,22 @@ bool CFileItem::LoadDetails()
     return false;
   }
 
+  if (IsVideo())
+  {
+    if (HasVideoInfoTag())
+      return true;
+
+    CVideoDatabase db;
+    if (!db.Open())
```
Comment (enen92) [inline xbmc/FileItem.cpp:3775]: maybe log error here? I think this is pretty much unexpected

Comment (ksooo) [reply xbmc/FileItem.cpp:3775]: Yes, can do.

File: xbmc/utils/ExecString.cpp [hunk 2]
```diff
@@ -33,6 +33,25 @@ CExecString::CExecString(const std::string& function, const std::vector<std::str
     SetExecString();
 }
 
+CExecString::CExecString(const std::string& function,
+                         const CFileItem& target,
+                         const std::string& param)
+  : m_function(function)
+{
+  m_valid = !m_function.empty() && !target.GetPath().empty();
```
[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/utils/ExecString.cpp:41]: It's path, because not content, but identity is needed here.

**Comment by ksooo** [comment]
@enen92 requested change to log db open error is in as separate commit. Good to go in now?

File: xbmc/FileItem.cpp [hunk 3]
```diff
@@ -3733,11 +3736,8 @@ bool CFileItem::LoadDetails()
                          this);
     }
     else
-    {
-      db.Close();
       return false;
-    }
-    db.Close();
```
Comment (enen92) [inline xbmc/FileItem.cpp:3740]: is this removal intentional? You don't seem to close the connection anymore

File: xbmc/FileItem.cpp [hunk 4]
```diff
@@ -3766,6 +3766,25 @@ bool CFileItem::LoadDetails()
...
+
+    CVideoDatabase db;
+    if (!db.Open())
+    {
+      CLog::LogF(LOGERROR, "Error opening video database");
+      return false;
+    }
+
+    if (db.LoadVideoInfo(GetDynPath(), *GetVideoInfoTag()))
+      return true;
```
Comment (enen92) [inline xbmc/FileItem.cpp:3782]: you seem to return here and in line 3785 without closing the db connection too. Is this intentional?

Comment (enen92) [reply xbmc/FileItem.cpp:3782]: Also on the last line too

File: xbmc/FileItem.cpp (hunk 3 above)
Comment (ksooo) [reply xbmc/FileItem.cpp:3740]: Yes, it is. See commit message for explanation.

File: xbmc/FileItem.cpp (hunk 4 above)
Comment (ksooo) [reply xbmc/FileItem.cpp:3782]: See my other comment. This is intended because the close call will be done anyway in the dtor of CVideoDatabase.

**Review by enen92 — APPROVED** [review]

**Comment by kraaa** [comment]
Not 100% sure, but pretty confident this as default action is not a good mix with skin playlists of tv-shows...

**Comment by ksooo** [comment]
> Not 100% sure, but pretty confident this as default action is not a good mix with skin playlists of tv-shows...

Says nothing to me, sorry. Could you please explain what you mean exactly?

**Comment by kraaa** [comment]
I'm writing a more detailed issue but:

I got skin.playlist Tv-shows watched lately as a background. 
When i select it and press enter/play/select it I get two options (Continue watching OR Restart watching) when before pressing enter it straight away entered the root of the show viewed in library where i could choose to resume or select another episode or whatever.
I can't find any options for default action either so i can make it like it was before.

**Comment by ksooo** [comment]
Please wait for the next nigthly build and report back whether the issue is fixed there. Could be that you were just hit by https://github.com/xbmc/xbmc/issues/23868 , the respective fix has just been merged.

**Comment by kraaa** [comment]
> Please wait for the next nigthly build and report back whether the issue is fixed there. Could be that you were just hit by #23868 , the respective fix has just been merged.

Oh yeah that's it. Thanks and sorry.
